src/proposals/proposals.controller.ts

```diff
@@ -578,7 +579,10 @@ export class ProposalsController {
     const proposal = await this.proposalsService.findOne({
       proposalId,
     });
-    if (!proposal) return { canAccess: false };
+
+    if (!proposal) {
+      throw new NotFoundException(`Proposal with ${proposalId} not found`);
+    }
```

Comment:
This endpoint is to avoid error throwing to the frontend for users who have not permission to access proposal. 
It should only return canAccess true or false, and additional information should just log with `logInfo` without throwing error.

Reply:
I understand that and it should work exactly as you described. Even if you don't have access to the proposal as you can see the `permissionChecker` is right after the proposal query. For example this throws error only if you provide a `proposalId` that doesn't exist. I think in this situation it is good to throw error. Otherwise after that where we check the permission we will return `true` or `false` accordingly. For me this is the best and most reasonable way. I don't see a point returning `200` success with `canAccess: false` if the queried proposal doesn't exist.

Reply:
At this moment, I do not know which is the best and most sensible way, although I would just return false if the proposal does not exists.
What I suggest is that both endpoints `proposal/{id}` and `proposal/{id}/authorization` behave the same way.

Reply:
Yes I understand you but then maybe it is better to just have the `permissionChecker` function to decide about this and not even have this condition in the `findByIdAccess` function. I will submit new improved solution and you can check it if you like it or not.

Reply:
The endpoint is designed to return only true or false to keep the frontend unaware of backend details. It’s used exclusively on the dataset details page on frontend , so we avoid throwing errors and simply show nothing if access fails for any reason, however, logging error to the grayLog would help us monitoring the issues.  
Ideally, error handling should occur within findById, not findByIdAccess. I don't know which way is better, what you suggested is good practice, I'm not sure if it's what we want

Reply:
That being said, if you can convince Max, I'm okay with this approach, as it is just to handle edge cases

Reply:
@Junjiequan this is being fixed by me and refactored a bit. You are welcome to double check it once more.